packages/taiko-client/driver/chain_syncer/event/syncer.go

```diff
@@ -440,11 +440,18 @@ func (s *Syncer) processShastaProposal(
 		); err != nil {
 			return fmt.Errorf("failed to insert Shasta blocks: %w", err)
 		}
-		if parent, err = s.rpc.L2.BlockByNumber(ctx, new(big.Int).Add(parent.Number(), common.Big1)); err != nil {
-			return fmt.Errorf("failed to fetch the new parent block: %w", err)
+		if parent, err = s.rpc.WaitL2Block(ctx, new(big.Int).Add(parent.Number(), common.Big1)); err != nil {
+			log.Warn("Failed to fetch the new parent block, keep retrying", "error", err)
```

Comment:
We are waiting for `new(big.Int).Add(parent.Number(), common.Big1)`, not `parent block` in this log, right?